### PR TITLE
#165305735 List Bookmarks

### PR DESCRIPTION
**Pull request by Philipotieno**
#### What does this PR do?
- List bookmarks of a logged in user
#### Description of Task to be completed?
- A logged in user should be able to see articles he/she bookmarked and be able to read them on clicking.
#### How should this be manually tested?
- `$ git clone https://github.com/andela/ah-the-immortals-frontend`
- `$ git checkout ft-list-bookmarks-165305735`
- `$ npm install`
- `$ npm run start:dev`
#### What are the relevant pivotal tracker stories?
[`#165305735`](https://www.pivotaltracker.com/n/projects/2326460/stories/165305735)
#### Screenshots
image

### CODE REVIEW

**Comment by matthenge** [comment]
@Philipotieno the cards listing the bookmarked articles should be placed at least 3 in each row, not 1. It should be consistent with the user's articles

**Comment by Philipotieno** [comment]
@matthenge Thanks for the feedback I have fixed it

File: src/redux/actions/__tests__/ListBookmarks.actions.test.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        data: {
+          bookmarks: [
+            {
+              'slug': 'tunechi-slug',
+              'username': 'tunechi',
+              'first_name': '',
+              'last_name': '',
+              'bio': '',
+              'img_url': 'https://res.cloudinary.com/grean//upload/v1560448760/57028948_10161507756365548_4757796820107657216_o_dtnz3g',
+              'created_at': '2019-04-30 11:52:18',
```
Comment (IssaIan) [inline src/redux/actions/__tests__/ListBookmarks.actions.test.js]: Remove the time here

**Review by IssaIan — CHANGES_REQUESTED** [review]

File: src/redux/actions/listBookmarks.actions.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  type: LIST_BOOKMARKS_FAILURE,
+  payload: error,
+});
+
+const listBookmarks = () => async (dispatch) => {
+  try {
+    const data = await listBookmarksApi();  
+    dispatch(listSuccess(data.data.articles));
+  } catch (error) {
+    dispatch(listFailure(error.response.data.message));
```
Comment (chariss-chomba-zetta) [inline src/redux/actions/listBookmarks.actions.js]: Catch network errors, eg if there is no internet `error.response.data.message` will be undefined

**Review by chariss-chomba-zetta — CHANGES_REQUESTED** [review]

File: src/redux/actions/__tests__/ListBookmarks.actions.test.js (hunk 1 above)
Comment (Philipotieno) [reply src/redux/actions/__tests__/ListBookmarks.actions.test.js]: Done

File: src/redux/actions/listBookmarks.actions.js (hunk 2 above)
Comment (Philipotieno) [reply src/redux/actions/listBookmarks.actions.js]: Okay working on that

**Comment by matthenge** [comment]
@Philipotieno the article cards are not consistent, some are bigger than others. Ensure they are of the same size. I also think that having 3 in each row will be better.

File: src/redux/actions/listBookmarks.actions.js (hunk 2 above)
Comment (Philipotieno) [reply src/redux/actions/listBookmarks.actions.js]: I have implemented that

Comment (IssaIan) [reply src/redux/actions/listBookmarks.actions.js]: Cool.

**Review by IssaIan — APPROVED** [review]

**Review by Amoswachira — APPROVED** [review]

**Pull request merged**